lib/script.js

```diff
@@ -28,23 +40,70 @@ function ShowCurrentTime() {
 } 
 
 function getWeather(city, state) {
-    fetch(`https://rocky-cliffs-95700.herokuapp.com/api/v1/forecast?location=${city,state}`)
+    // if the city/state combination is already stored, find and use it instead of making a new call. else, make a new call.
+    fetch(`https://rocky-cliffs-95700.herokuapp.com/api/v1/forecast?location=${city},${state}`)
     .then(response => response.json())
-    .then(result => weather_data.push(result));
+    .then(result => weather_data.unshift(result));
```

Comment:
If this is the only element you are adding to the empty array, you could use a block here instead and just assign the result to a variable, instead of adding it to an array.
.then(response => {
assign variable here
          })

Reply:
I would love to implement it that way...I still have trouble understanding Promises and arrow functions at the moment and found that variable assignment was easier than trying to get this function to return the api response as an object. I tried for hours and hours....